app/src/main/java/com/wajahatkarim3/imagine/base/BaseFragment.kt

```diff
@@ -1,11 +1,27 @@
 package com.wajahatkarim3.imagine.base
 
-import android.content.Context
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
 import androidx.fragment.app.Fragment
+import androidx.viewbinding.ViewBinding
 
-abstract class BaseFragment: Fragment() {
+abstract class BaseFragment<VB : ViewBinding> : Fragment() {
 
-    override fun onAttach(context: Context) {
-        super.onAttach(context)
+    private var _binding: VB? = null
+    protected val binding: VB get() = _binding!!
```

Comment:
Can you rename this to `bi` instead of `binding`. The whole app uses `bi`, so this would be more consistent than renaming everywhere.